### ISSUE TITLE
UI doesn't show any runs while "aim runs ls" does

### ISSUE THREAD

**Issue by andrey-klochkov-liftoff** [aimhubio/aim#2510]
## 🐛 Bug

I track runs in a repository on the local disk initialized with `aim init --repo /repo/dir`. If I do `aim runs --repo /repo/dir ls`, it prints run hashes, all the runs are there. I then start the UI with `aim ui --repo /repo/dir`, but the run page shows "No Results". How can I troubleshoot this?

I used aim successfully before with the same exact code, same aim version, and it worked OK. Restarting UI doesn't help, and the UI CLI doesn't seem to have any kind of verbose logging.

### Environment

- Aim Version 3.14.3
- Python version 3.9.10
- pip version 21.3.1
- OS Ubuntu 20.04
- Any other relevant information

Logs from the training process that are from aim:
```
2023-01-29 07:50:37.503 INFO __init__ - __init__: creating RunStatusReporter for <Run#-8889999695818297307 name=4ab9d76b5bae4049b683b62c repo=<Repo#-6761918083593033162 path=/repo/dir/.aim read_only=None>>
2023-01-29 07:50:37.504 INFO __init__ - __init__: polling for check-ins in /repo/dir/.aim/check_ins
2023-01-29 07:50:37.506 INFO __init__ - poll: no check-in found for 4ab9d76b5bae4049b683b62c; returning zero-check-in
2023-01-29 07:50:37.506 INFO __init__ - __init__: no leftover check-in found. starting from zero
2023-01-29 07:50:37.506 INFO __init__ - increment: incrementing check-in: CheckIn(idx=0, expect_next_in=0, flag_name='check_in') -> CheckIn(idx=1, expect_next_in=0, flag_name='starting')
2023-01-29 07:50:37.506 INFO __init__ - touch: touching check-in: /repo/dir/.aim/check_ins/4ab9d76b5bae4049b683b62c-0000000000000001-starting-1674978637.51-00000
2023-01-29 07:50:37.508 INFO __init__ - cleanup: found 0 check-ins:
2023-01-29 07:50:37.508 INFO __init__ - cleanup: the acting one: /repo/dir/.aim/check_ins/4ab9d76b5bae4049b683b62c-0000000000000001-starting-1674978637.51-00000
2023-01-29 07:50:37.508 INFO __init__ - first_seen_cached: * first time (1322059.574321339) seen for (1, 0, 'starting') for 4ab9d76b5bae4049b683b62c
2023-01-29 07:50:37.508 INFO __init__ - cleanup: returning acting check-in after cleanup: CheckIn(idx=1, expect_next_in=0, flag_name='starting')
2023-01-29 07:50:37.508 INFO __init__ - __init__: starting from: CheckIn(idx=1, expect_next_in=0, flag_name='starting')
2023-01-29 07:50:37.508 INFO __init__ - __init__: starting writer thread for <aim.sdk.reporter.RunStatusReporter object at 0x7f7314aac520>

...

2023-01-29 07:51:08.916 INFO train - <module>: Model saved to /models/4ab9d76b5bae4049b683b62c/model.onnx
2023-01-29 07:51:08.916 INFO __init__ - increment: incrementing check-in: CheckIn(idx=1, expect_next_in=0, flag_name='starting') -> CheckIn(idx=2, expect_next_in=0, flag_name='finished')
2023-01-29 07:51:08.916 INFO __init__ - flush: notifying <aim.sdk.reporter.RunStatusReporter object at 0x7f7314aac520>
2023-01-29 07:51:08.916 INFO __init__ - flush: blocking until the writer finishes
2023-01-29 07:51:08.917 INFO __init__ - writer: detected newest check-in: CheckIn(idx=2, expect_next_in=0, flag_name='finished')
2023-01-29 07:51:08.917 INFO __init__ - touch: touching check-in: /repo/dir/.aim/check_ins/4ab9d76b5bae4049b683b62c-0000000000000002-finished-1674978668.92-00000
2023-01-29 07:51:08.920 INFO __init__ - cleanup: found 0 check-ins:
2023-01-29 07:51:08.920 INFO __init__ - cleanup: the acting one: /repo/dir/.aim/check_ins/4ab9d76b5bae4049b683b62c-0000000000000002-finished-1674978668.92-00000
2023-01-29 07:51:08.920 INFO __init__ - first_seen_cached: * first time (1322090.986186676) seen for (2, 0, 'finished') for 4ab9d76b5bae4049b683b62c
2023-01-29 07:51:08.920 INFO __init__ - cleanup: returning acting check-in after cleanup: CheckIn(idx=2, expect_next_in=0, flag_name='finished')
2023-01-29 07:51:08.920 INFO __init__ - writer: changing to -> CheckIn(idx=2, expect_next_in=0, flag_name='finished')
2023-01-29 07:51:09.845 INFO __init__ - flush: notifying <aim.sdk.reporter.RunStatusReporter object at 0x7f7314aac520>
2023-01-29 07:51:09.845 INFO __init__ - writer: writer thread stopping as requested
```

Logs from "aim ui" with `--log-level DEBUG`:
```
DEBUG:filelock:Attempting to acquire lock 140463952245952 on /repo/dir/.aim/.repo_lock
DEBUG:filelock:Lock 140463952245952 acquired on /repo/dir/.aim/.repo_lock
DEBUG:filelock:Attempting to release lock 140463952245952 on /repo/dir/.aim/.repo_lock
DEBUG:filelock:Lock 140463952245952 released on /repo/dir/.aim/.repo_lock
INFO  [sqlalchemy.engine.Engine] BEGIN (implicit)
INFO  [sqlalchemy.engine.Engine] PRAGMA main.table_info("alembic_version")
INFO  [sqlalchemy.engine.Engine] [raw sql] ()
INFO  [sqlalchemy.engine.Engine] SELECT alembic_version.version_num 
FROM alembic_version
INFO  [sqlalchemy.engine.Engine] [generated in 0.00011s] ()
INFO  [sqlalchemy.engine.Engine] ROLLBACK
┌------------------------------------------------------------------------┐
                Aim UI collects anonymous usage analytics.                
                        Read how to opt-out here:                         
    https://aimstack.readthedocs.io/en/latest/community/telemetry.html    
└------------------------------------------------------------------------┘
Running Aim UI on repo `<Repo#993126864044414543 path=/repo/dir/.aim read_only=None>`
Open http://0.0.0.0:43800
Press Ctrl+C to exit
INFO:     Started server process [2149840]
INFO:     Waiting for application startup.
INFO:     Application startup complete.
INFO:     Uvicorn running on http://0.0.0.0:43800 (Press CTRL+C to quit)
2023-01-29 08:03:17,190 INFO sqlalchemy.engine.Engine BEGIN (implicit)
2023-01-29 08:03:17,196 INFO sqlalchemy.engine.Engine SELECT run.id AS run_id, run.hash AS run_hash, run.name AS run_name, run.description AS run_description, run.is_archived AS run_is_archived, run.created_at AS run_created_at, run.updated_at AS run_updated_at, run.finalized_at AS run_finalized_at, run.experiment_id AS run_experiment_id, experiment_1.id AS experiment_1_id, experiment_1.uuid AS experiment_1_uuid, experiment_1.name AS experiment_1_name, experiment_1.is_archived AS experiment_1_is_archived, experiment_1.created_at AS experiment_1_created_at, experiment_1.updated_at AS experiment_1_updated_at, tag_1.id AS tag_1_id, tag_1.uuid AS tag_1_uuid, tag_1.name AS tag_1_name, tag_1.color AS tag_1_color, tag_1.description AS tag_1_description, tag_1.is_archived AS tag_1_is_archived, tag_1.created_at AS tag_1_created_at, tag_1.updated_at AS tag_1_updated_at 
FROM run LEFT OUTER JOIN experiment AS experiment_1 ON experiment_1.id = run.experiment_id LEFT OUTER JOIN (run_tag AS run_tag_1 JOIN tag AS tag_1 ON tag_1.id = run_tag_1.tag_id) ON run.id = run_tag_1.run_id ORDER BY run.created_at DESC
2023-01-29 08:03:17,196 INFO sqlalchemy.engine.Engine [generated in 0.00015s] ()
INFO:     172.20.91.178:63949 - "GET /runs/search/run?q=&limit=45&report_progress=False HTTP/1.1" 200 OK
2023-01-29 08:03:27,226 INFO sqlalchemy.engine.Engine SELECT run.id AS run_id, run.hash AS run_hash, run.name AS run_name, run.description AS run_description, run.is_archived AS run_is_archived, run.created_at AS run_created_at, run.updated_at AS run_updated_at, run.finalized_at AS run_finalized_at, run.experiment_id AS run_experiment_id, experiment_1.id AS experiment_1_id, experiment_1.uuid AS experiment_1_uuid, experiment_1.name AS experiment_1_name, experiment_1.is_archived AS experiment_1_is_archived, experiment_1.created_at AS experiment_1_created_at, experiment_1.updated_at AS experiment_1_updated_at, tag_1.id AS tag_1_id, tag_1.uuid AS tag_1_uuid, tag_1.name AS tag_1_name, tag_1.color AS tag_1_color, tag_1.description AS tag_1_description, tag_1.is_archived AS tag_1_is_archived, tag_1.created_at AS tag_1_created_at, tag_1.updated_at AS tag_1_updated_at 
FROM run LEFT OUTER JOIN experiment AS experiment_1 ON experiment_1.id = run.experiment_id LEFT OUTER JOIN (run_tag AS run_tag_1 JOIN tag AS tag_1 ON tag_1.id = run_tag_1.tag_id) ON run.id = run_tag_1.run_id ORDER BY run.created_at DESC
2023-01-29 08:03:27,226 INFO sqlalchemy.engine.Engine [cached since 10.03s ago] ()
INFO:     172.20.91.178:63950 - "GET /runs/search/run?q=&limit=45&report_progress=False HTTP/1.1" 200 OK
```

**Comment by alberttorosyan** [comment]
@andrey-klochkov-liftoff, is it possible that the runs are marked as archived? You can check that by querying runs with
```run.archived == True```

**Comment by nlgranger** [comment]
Same issue here. 

<details>
  <summary>environement.txt</summary>

```
absl-py==1.3.0
aim==3.15.2
aim-ui==3.15.2
aimrecords==0.0.7
aimrocks==0.2.1
aiofiles==22.1.0
alembic==1.9.2
base58==2.0.1
cachetools==5.2.0
certifi==2022.9.24
cffi==1.15.1
charset-normalizer==2.1.1
click==8.1.3
contourpy==1.0.6
cryptography==39.0.0
cycler==0.11.0
fastapi==0.67.0
filelock==3.9.0
fonttools==4.38.0
google-auth==2.15.0
google-auth-oauthlib==0.4.6
greenlet==2.0.2
grpcio==1.51.1
h11==0.14.0
idna==3.4
ijson==3.2.0
Jinja2==3.1.2
joblib==1.2.0
kiwisolver==1.4.4
Mako==1.2.4
Markdown==3.4.1
MarkupSafe==2.1.1
matplotlib==3.6.2
numpy==1.23.5
oauthlib==3.2.2
packaging==21.3
pandas==1.5.2
Pillow==9.3.0
Pillow-SIMD==9.0.0.post1
protobuf==3.20.3
psutil==5.9.4
py3nvml==0.2.7
pyasn1==0.4.8
pyasn1-modules==0.2.8
pycocotools==2.0.6
pycparser==2.21
pydantic==1.10.4
pyparsing==3.0.9
python-dateutil==2.8.2
pytz==2022.6
requests==2.28.1
requests-oauthlib==1.3.1
RestrictedPython==6.0
rsa==4.9
scikit-learn==1.1.3
scipy==1.9.3
six==1.16.0
SQLAlchemy==2.0.0
starlette==0.14.2
tensorboard==2.11.0
tensorboard-data-server==0.6.1
tensorboard-plugin-wit==1.8.1
threadpoolctl==3.1.0
torch==1.13.0+cu116
torch-tb-profiler==0.4.0
torchvision==0.14.0+cu116
tqdm==4.64.1
typing_extensions==4.4.0
urllib3==1.26.13
uvicorn==0.20.0
Werkzeug==2.2.2
xmltodict==0.13.0
```

</details>

**Comment by CaptainTrojan** [comment]
I suffer from this issue as well. Same Aim and python version as @nlgranger. I can confirm that, for myself, the runs are not marked as archived. Tested this with both remote and local tracking.

**Comment by zombie-einstein** [comment]
Think I'm also seeing the same issue. From a fresh install of aim can see runs with `aim runs ls` and via API but UI always shows "No Results".

- Ubuntu 22.04
- Python 3.10
- Aim 3.15.2

Though I did try out some downgraded versions, and saw the same behaviour.

**Comment by whlteXbread** [comment]
there was some discussion about this in the discord—seemed like there was some issue with `sqlalchemy==2.0.0`—if you downgrade (`pip install 'sqlalchemy<2.0.0'`) new installs will work fine.

there's a solution for fixing existing repos in the discord too, but i didn't want to share the solution outside of the discord without permission

**Comment by andrey-klochkov-liftoff** [comment]
@whlteXbread ,
Thanks, downgrading `sqlalchemy` and recreating the repo has fixed it for me. It's still just a workaround, I'll keep this bug report open.

**Comment by alberttorosyan** [comment]
The temporary fix (setting upper limit for `sqlalchemy` version) is ready: https://github.com/aimhubio/aim/pull/2513
We'll keep looking further into this to see what changes in `sqlalchemy 2.0.0` broke things on our side. Meanwhile, in order to fix aim repo with this issue you can do the following:

1. downgrade sqlalchemy - pip install sqlalchemy<2.0.0
2. Make sure there are no active runs
3. Run this simple script to reopen the previous runs, so then they'll have the sufficient data to appear on UI:
```
from aim import Run, Repo
from aim.sdk.lock_manager import LockManager

repo = Repo.from_path({your_repo_path})
run_hashes = repo.list_all_runs()

lock_manager = LockManager(repo.path)
for run_hash in run_hashes:
    lock_manager.release_locks(run_hash, force=True)
    run = Run(run_hash, read_only=False)
    run.close()
```

**Comment by mihran113** [comment]
Hey folks, the fix of this issue has been shipped with `3.16.0` release. Let us know if everything works as expected.

**Comment by gorarakelyan** [comment]
Closing the issue, as the fix has been shipped. Please feel free to reopen this, if there are any issues still.
